### PR TITLE
Bring edu benefits form up to date with wireframe

### LINKED ISSUE #3056: Update existing pages to match latest wireframes
Some of our form pages are likely out of date now, we should make another pass through each one and make any necessary updates.

### PR DESCRIPTION
**Pull request by jbalboni**
Closes #3056.

This has several small updates to the form to bring it in line with changes to the wireframe. There are probably copy differences that still exist and those will be made separately.

The json schema has been updated with the changes, as well.

### CODE REVIEW

**Comment by jbalboni** [comment]
@alexose Want to take a look?

File: src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx [hunk 1]
```diff
@@ -20,11 +22,31 @@ export default class BenefitsSelectionFields extends React.Component {
               options={relinquishableBenefits}
               value={this.props.data.benefitsRelinquished}
               onValueChange={(update) => {this.props.onStateChange('benefitsRelinquished', update);}}/>
+          {showRelinquishedEffectiveDate(this.props.data.benefitsRelinquished.value)
+            ? <DateInput required={showRelinquishedEffectiveDate(this.props.data.benefitsRelinquished.value)}
+                errorMessage="Please provide a response"
+                validation={validateIfDirtyDateObj(this.props.data.benefitsRelinquishedDate, isValidDateField)}
+                label="Effective date"
```
Comment (akainic) [inline src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx]: Should we have a more descriptive label than just "Effective date"? Like "Effective date of relinquished benefit" or something? @cweber @goldenmeanie

File: src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx [hunk 2]
```diff
@@ -20,11 +22,31 @@ export default class BenefitsSelectionFields extends React.Component {
               options={relinquishableBenefits}
               value={this.props.data.benefitsRelinquished}
               onValueChange={(update) => {this.props.onStateChange('benefitsRelinquished', update);}}/>
+          {showRelinquishedEffectiveDate(this.props.data.benefitsRelinquished.value)
+            ? <DateInput required={showRelinquishedEffectiveDate(this.props.data.benefitsRelinquished.value)}
+                errorMessage="Please provide a response"
+                validation={validateIfDirtyDateObj(this.props.data.benefitsRelinquishedDate, isValidDateField)}
```
Comment (akainic) [inline src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx]: Why was this added?

**Comment by alexose** [comment]
Lookin'!

**Review by akainic — COMMENTED** [review]
Lots of changes here! Wanted to check this out locally but the application isn't loading. Thoughts?

File: src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx (hunk 2 above)
Comment (jbalboni) [inline src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx]: It's a new field in the wireframe, I'm not positive on why it's needed.

File: src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx (hunk 1 above)
Comment (jbalboni) [inline src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx]: I have been assuming that things like this are going to change when we get stuff back from the content people.

File: src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx (hunk 2 above)
Comment (cweber) [inline src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx]: It's a required element to indicate when you're benefits you're currently receiving will get shot off in favor of the new benefits through CH 33.

File: src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx (hunk 1 above)
Comment (cweber) [inline src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionFields.jsx]: Agreed and correct...though this is one of the things that they missed in what they delivered. Should have it soon though.

File: src/js/edu-benefits/components/veteran-information/VeteranInformationFields.jsx [hunk 3]
```diff
@@ -24,9 +26,12 @@ export default class PersonalInformationFields extends React.Component {
               month={this.props.data.veteranDateOfBirth.month}
               year={this.props.data.veteranDateOfBirth.year}
               onValueChange={(update) => {this.props.onStateChange('veteranDateOfBirth', update);}}/>
-          <Gender
+          <ErrorableRadioButtons
```
Comment (jbalboni) [inline src/js/edu-benefits/components/veteran-information/VeteranInformationFields.jsx]: @akainic I changed Gender to this because @cweber and @goldenmeanie wanted the gender field to use radio buttons. I could also just change the Gender component. Thoughts?

File: src/js/edu-benefits/components/military-history/RotcScholarship.jsx [hunk 4]
```diff
@@ -9,25 +9,29 @@ export default class RotcScholarship extends React.Component {
...
+              onValueChange={(update) => {onValueChange('year', update);}}/>
+        </div>
+        <div className="columns small-8">
+          <ErrorableTextInput
+              errorMessage={validateIfDirty(scholarship.amount, isNotBlank) ? undefined : 'Please enter the amount received.'}
+              validation={validateIfDirty(scholarship.amount, isNotBlank)}
+              label="Scholarship amount"
+              name="amount"
+              field={scholarship.amount}
+              placeholder="$"
```
Comment (akainic) [inline src/js/edu-benefits/components/military-history/RotcScholarship.jsx]: Do you think people will interpret this as needing a `$` in front of their input? Like entering `$10` instead of just `10`? I'm assuming this is not required.

File: src/js/edu-benefits/components/veteran-information/VeteranInformationFields.jsx (hunk 3 above)
Comment (akainic) [inline src/js/edu-benefits/components/veteran-information/VeteranInformationFields.jsx]: Hm good question. I would keep it as you have it so we can keep the dropdown for HCA (although not sure why we need a dropdown for HCA... maybe something to revisit later :)).

**Review by akainic — COMMENTED** [review]
LGTM with a couple of small questions!

File: src/js/edu-benefits/components/military-history/RotcScholarship.jsx (hunk 4 above)
Comment (jbalboni) [inline src/js/edu-benefits/components/military-history/RotcScholarship.jsx]: That's a good question. It works either way and this is the placeholder in the wireframes.

Comment (jbalboni) [inline src/js/edu-benefits/components/military-history/RotcScholarship.jsx]: Actually, it wasn't working if you did $10. Fixed now.
